### ISSUE TITLE
Fix `MD041` false positive for leading MkDocs anchor line `[](){ #... }`

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#365]
## Problem

`rumdl` reports `[MD041] First line in file should be a level 1 heading` when a Markdown file starts with a MkDocs-style anchor line like `[](){ #example }`, even if the next line is a valid `#` heading.

## Steps to reproduce

1. Create `test.md` with:

   ```markdown
   [](){ #example }
   # Title
   ```

2. Run `rumdl` on the file (for example: `rumdl test.md`).

## Expected

`MD041` should ignore a leading "anchor-only" line like `[](){ #example }` and treat `# Title` as the first content line, so no `MD041` warning is emitted.

## Actual

`MD041` is emitted because the first line is not a level 1 heading.

## Suggested fix

When evaluating `MD041`, skip leading lines that match the MkDocs anchor pattern (for example `[](){ #id }`, possibly allowing whitespace variations) before checking for the first required `#` heading.

**Comment by rvben** [comment]
This was fixed in v0.1.10 (commit f90bd34e). The fix requires `flavor = "mkdocs"` in your config:

```toml
[global]
flavor = "mkdocs"
```

Thanks for reporting @pygarap!
